ui/src/components/credentials/LinksTable.tsx

```diff
@@ -0,0 +1,401 @@
+                },
+                {
+                  key: "divider",
+                  type: "divider",
+                },
+                {
+                  danger: true,
+                  icon: <IconTrash />,
+                  key: "delete",
+                  label: "Delete",
+                },
+              ],
+            }}
+          >
+            <Row>
+              <IconDots className="icon-secondary" />
+            </Row>
+          </Dropdown>
+        </Row>
+      ),
+      sorter: ({ status: a }, { status: b }) => a.localeCompare(b),
+      title: "Status",
+      width: 140,
+    },
+  ];
+
+  const fetchLinks = useCallback(
+    async (signal: AbortSignal) => {
+      setLinks((previousLinks) =>
+        isAsyncTaskDataAvailable(previousLinks)
+          ? { data: previousLinks.data, status: "reloading" }
+          : { status: "loading" }
+      );
+
+      const response = await getLinks({
+        env,
+        params: {
+          query: queryParam || undefined,
+          status: status,
+        },
+        signal,
+      });
+
+      if (response.isSuccessful) {
+        setLinks({ data: response.data, status: "successful" });
+      } else {
+        if (!isAbortedError(response.error)) {
+          setLinks({ error: response.error, status: "failed" });
+        }
+      }
+    },
+    [env, queryParam, status]
+  );
+
+  const handleStatusChange = ({ target: { value } }: RadioChangeEvent) => {
+    const parsedLinkValue = linkStatusParser.safeParse(value);
+
+    if (parsedLinkValue.success) {
+      const params = new URLSearchParams(searchParams);
+
+      if (parsedLinkValue.data !== "all") {
+        params.set(STATUS_SEARCH_PARAM, parsedLinkValue.data);
+      } else {
+        params.delete(STATUS_SEARCH_PARAM);
+      }
+
+      setLinks({ status: "pending" });
+      setSearchParams(params);
+    } else {
+      processZodError(parsedLinkValue.error).forEach((error) => void message.error(error));
+    }
+  };
+
+  const updateCredentialInState = (active: Link["active"], id: Link["id"]) => {
+    setLinks((oldLinks) =>
+      isAsyncTaskDataAvailable(oldLinks)
+        ? {
+            data: oldLinks.data.reduce((links: Link[], currentLink: Link) => {
+              if (currentLink.id === id) {
+                if (status === currentLink.status) {
+                  return links;
+                } else {
+                  const linkStatusInverted: Link = {
+                    ...currentLink,
+                    active,
+                    status: currentLink.status === "active" ? "inactive" : "active",
+                  };
+
+                  return [...links, linkStatusInverted];
+                }
+              } else {
+                return [...links, currentLink];
+              }
+            }, []),
+            status: "successful",
+          }
+        : oldLinks
+    );
+  };
+
+  const toggleLinkActive = (active: boolean, id: Link["id"]) => {
```

Comment:
Any reason why this separate issue's functionality was added as part of this PR?